core/core-funcs.el

```diff
@@ -36,12 +36,18 @@
   (run-hooks 'text-mode-hook))
 
 (defun spacemacs//get-package-directory (pkg)
-  "Return the directory of PKG. Return nil if not found."
-  (let ((elpa-dir (file-name-as-directory package-user-dir)))
-    (when (file-exists-p elpa-dir)
-      (let* ((pkg-match (concat (symbol-name pkg) "-[0-9]+"))
-             (dir (car (directory-files elpa-dir 'full pkg-match))))
-        (when dir (file-name-as-directory dir))))))
+  "Return the installed directory of PKG. Return `nil' if not found."
+  (let ((directories (cons package-user-dir
```

Comment:
There is nothing wrong with site-lisp.  In fact I want to put installed packaged under site-lisp.  With the proposed code, I do have the installed files under site-lisp.  The issue is how does spacemacs know that the packages are installed already so that it does not install them under package-user-dir?

AFAIK package-directory-list was created to solve this problem, i.e., to allow site administrators to install packages under site-lisp directories.